src/main/java/ezuino/Main.java

```diff
@@ -79,23 +83,27 @@ private static void contextualAnalysis(AstNode ast, ErrorHandler errorHandler) {
         ast.acceptLevel(indentedPrintVisitor, 0);
 
         boolean printDcl = true;
+        boolean hasNoError = !errorHandler.hasErrors();
 
-        ast.accept(new SymbolTableVisitor(printDcl, errorHandler));
+        if(hasNoError) {
+            ast.accept(new SymbolTableVisitor(printDcl, errorHandler));
+        }
         ast.acceptLevel(indentedPrintVisitor, 0);
-
-        ast.accept(new Typechecker(errorHandler));
         ast.acceptLevel(indentedPrintVisitor, 0);
-
-        ast.accept(new ReturnStmtTypeCheckVisitor(errorHandler));
-
-        ast.accept(new MissingReturnStmtVisitor(errorHandler));
-
-        ast.accept(new FuncStructureVisitor(errorHandler));
+        hasNoError = !errorHandler.hasErrors();
+        if(hasNoError) {
+            ast.accept(new Typechecker(errorHandler));
+            ast.accept(new ReturnStmtTypeCheckVisitor(errorHandler));
+            ast.accept(new MissingReturnStmtVisitor(errorHandler));
+            ast.accept(new FuncStructureVisitor(errorHandler));
+        }
     }
 
-    private static void codeGeneration(AstNode ast) {
-
-        ast.accept(new CCodeGenerationVisitor(System.out));
+    private static void codeGeneration(AstNode ast, ErrorHandler errorHandler) {
```

Comment:
Unused variable `errorHandler.
`ErrorHandler` is currently meant to be for reporting errors to the user (IE that SHE/HE fucked up). If there are problems in the code generation, it isn't the users fault, but our problem